### PR TITLE
add template for Visual Studio

### PR DESCRIPTION
**Pull request by erich-wang**

### CODE REVIEW

File: content-vs/dotnet-template-azure-iot-edge-module/.gitignore [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Ignore the following files which are copied from ROOT/content/dotnet-template-azure-iot-edge-module/CSharp
+
+CSharp/Dockerfile*
```
Comment (LazarusX) [inline content-vs/dotnet-template-azure-iot-edge-module/.gitignore]: From the perspective of users, these files should be version controlled.

File: content-vs/dotnet-template-azure-iot-edge-module/CSharp/module.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+    "$schema-version": "0.0.1",
+    "description": "",
+    "image": {
+        "repository": "<repository>",
```
Comment (LazarusX) [inline content-vs/dotnet-template-azure-iot-edge-module/CSharp/module.json]: Why is a copy of this file needed to be maintained here?

File: content/dotnet-template-azure-iot-edge-module/CSharp/SampleModule.csproj [hunk 3]
```diff
@@ -3,12 +3,16 @@
...
   </PropertyGroup>
-  
+
   <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|netcoreapp2.1|AnyCPU'">
     <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
     <TreatSpecificWarningsAsErrors />
   </PropertyGroup>
 
+  <ItemGroup>
+    <ProjectCapability Include="AzureIoTEdgeModule"/>
```
Comment (LazarusX) [inline content/dotnet-template-azure-iot-edge-module/CSharp/SampleModule.csproj]: Is there any side effect for users who don't use VS?

**Comment by LazarusX** [comment]
Please also update tests.

File: content-vs/dotnet-template-azure-iot-edge-module/.gitignore (hunk 1 above)
Comment (erich-wang) [reply content-vs/dotnet-template-azure-iot-edge-module/.gitignore]: this .ignore file is not used by user, it is just used to ignore files copied from content/

File: content-vs/dotnet-template-azure-iot-edge-module/CSharp/module.json (hunk 2 above)
Comment (erich-wang) [reply content-vs/dotnet-template-azure-iot-edge-module/CSharp/module.json]: arm32 platfrom is not included in this file

**Comment by erich-wang** [comment]
> Please also update tests.

There's no simple way to test vs project template. I prefer to do integration test with iot edge extension.

File: content-vs/dotnet-template-azure-iot-edge-module/CSharp/module.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        "tag": {
+            "version": "0.0.1",
+            "platforms": {
+                "amd64": "./Dockerfile.amd64", 
+                "amd64.debug": "./Dockerfile.amd64.debug",
+                "arm32v7": "./Dockerfile.arm32v7",
```
Comment (formulahendry) [inline content-vs/dotnet-template-azure-iot-edge-module/CSharp/module.json]: Remove?

File: content-vs/dotnet-template-azure-iot-edge-module/.gitignore (hunk 1 above)
Comment (LazarusX) [reply content-vs/dotnet-template-azure-iot-edge-module/.gitignore]: Will this `.gitignore` file be copied to users' project folder?

**Pull request closed without merging**